### ISSUE TITLE
Implement file hierarchy for collections

### ISSUE THREAD

**Issue by kwajiehao** [isomerpages/isomercms-frontend#204]
This issue is similar to #202 but it's complicated by the fact that collection pages can contain third nav sections. At the top-level collection page, we want to allow users to create a page, as well as assign it to a collection and a third nav within that collection. However, to do that, we would need to retrieve all pages from all collections to be able to identify all third navs which can be assigned. This is bad because it means the load speed will not improve from the current code.

Instead, we should implement a bunch of YAML files which store the structural contents of each collection. For example, a collection called `Collection A` would have a `_data/collections/collection_a.yml` file which contains:

```
pages:
  - type: page
     title: ABC
  - type: page
     title: DEF
  - type: third-nav
     title: Sample third-nav
     contents:
       - type: third-nav-page
          title: GHI
```

From these files, we would be able to determine which third nav sections exist. We can also implement them in a lazy manner - meaning, only retrieve this structure file when the user selects a collection to assign a page to.

**Comment by kwajiehao** [comment]
Reopened this issue because the third nav functionality has not yet been accomplished - we've only finished the visual elements of the solution described

**Comment by kwajiehao** [comment]
Resolved by #218
